platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1150,6 +1161,18 @@ - (id)mgl_jsonExpressionObject {
                 } else if ([type isEqualToString:@"NSNumber"]) {
                     return @[@"to-number", object];
                 }
+#if TARGET_OS_IPHONE
+                else if ([type isEqualToString:@"UIColor"] || [type isEqualToString:@"MGLColor"]) {
+                    return @[@"to-color", object];
+                }
+#else
+                else if ([type isEqualToString:@"NSColor"] || [type isEqualToString:@"MGLColor"]) {
+                    return @[@"to-color", object];
+                }
+#endif
+                else if ([type isEqualToString:@"NSArray"]) {
+                    return @[@"to-rgba", object];
```

Comment:
NSExpression is a bit aggressive in evaluating a constant in a cast even during parsing, so just initializing an NSExpression with a format string of `CAST(%@, 'NSArray')` would crash. However, a format string of `CAST(noindex(%@, 'NSArray')` should convert to the expected JSON without crashing.